scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,220 @@
+    if not success:
+        logging.error('Parsing suite definition file {0} failed.'.format(sdf))
+        success = False
+    return (success, suite)
+
+def create_metadata_filename_dict(metapath):
+    """Given a path, read all .meta files in that directory and add them to a dictionary: the keys
+       are the name of the scheme, and the values are the filename of the .meta file associated
+       with that scheme"""
+
+    success = True
+    metadata_dict = {}
+    scheme_filenames=glob.glob(metapath + "*.meta")
+    if not scheme_filenames:
+        logging.error(f'No files found in {metapath} with ".meta" extension')
+        success = False
+
+    for scheme_fn in scheme_filenames:
+        schemes=find_scheme_names(scheme_fn)
+        # The above returns a list of schemes in each filename, but
+        # we want a dictionary of schemes associated with filenames:
+        for scheme in schemes:
+            metadata_dict[scheme]=scheme_fn
+
+    return (metadata_dict, success)
+
+
+def create_var_graph(suite, var, config, metapath):
+    """Given a suite, variable name, a 'config' dictionary, and a path to .meta files:
+         1. Creates a dictionary associating schemes with their .meta files
+         2. Loops through the call tree of the provided suite
+         3. For each scheme, reads .meta file for said scheme, checks for variable within that
+            scheme, and if it exists, adds an entry to a list of tuples, where each tuple includes
+            the name of the scheme and the intent of the variable within that scheme"""
+
+    success = True
+
+    # Create a list of tuples that will hold the in/out information for each scheme
+    var_graph=[]
+
+    logging.debug("reading .meta files in path:\n {0}".format(metapath))
+    (metadata_dict, success)=create_metadata_filename_dict(metapath)
+    if not success:
+        raise Exception('Call to create_metadata_filename_dict failed')
+
+    logging.debug(f"reading metadata files for schemes defined in config file: "
+                  f"{config['scheme_files']}")
+
+    # Loop through call tree, find matching filename for scheme via dictionary schemes_in_files,
+    # then parse that metadata file to find variable info
+    partial_matches = {}
+    for scheme in suite.call_tree:
+        logging.debug("reading meta file for scheme {0} ".format(scheme))
+
+        if scheme in metadata_dict:
+            scheme_filename = metadata_dict[scheme]
+        else:
+            raise Exception(f"Error, scheme '{scheme}' from suite '{suite.sdf_name}' "
+                            f"not found in metadata files in {metapath}")
+
+        logging.debug("reading metadata file {0} for scheme {1}".format(scheme_filename, scheme))
+
+        new_metadata_headers = parse_metadata_file(scheme_filename, \
+                                                   known_ddts=registered_fortran_ddt_names(), \
+                                                   logger=logging.getLogger(__name__))
```

Comment:
The line continuations shouldn't be needed here for continuing arguments on the next line.